src/control/server/ctl_ranks_rpc.go

```diff
@@ -424,46 +422,45 @@ func (svc *ControlService) SetEngineLogMasks(ctx context.Context, req *ctlpb.Set
 		return nil, errors.New("nil request")
 	}
 
-	errs := []string{}
+	resp := new(ctlpb.SetLogMasksResp)
+	instances := svc.harness.Instances()
+	resp.Errors = make([]string, len(instances))
 
-	for idx, ei := range svc.harness.Instances() {
+	for idx, ei := range instances {
 		eReq := *req // local per-engine copy
 
+		if int(ei.Index()) != idx {
+			svc.log.Errorf("engine instance index %d doesn't match engine.Index %d",
+				idx, ei.Index())
```

Comment:
If this could happen, why not keep the "engine-[num]" string in the error messages? Just to ensure the user always ends up with correct information, even if something weird is happening.

Reply:
Good idea, I will update if I have to re-push, otherwise I will add as part of a subsequent PR. Could you please approve the PR unless this is a blocking comment? thanks in advance